goalgorilla/open_social
# Issue #3280216 by colan: "Please log in or sign up to comment" is displayed to authenticated users

## Problem

After upgrading to Open Social 11, authenticated users are seeing "Please log in or sign up to comment", which makes no sense because they're already logged in.

This is being caused by [the changes made](https://git.drupalcode.org/project/social/-/commit/1810027d848657251465b6ece6df1404ac4fd4d9) for [#3251540: Access to comment same as fo the commented entity](https://www.drupal.org/project/social/issues/3251540). It was assumed that only anonymous users shouldn't get the comment form, which doesn't take into account authenticated users who aren't in the group.

## Solution

There are several options if the user is already logged in:
1. Do nothing.
2. Display a more appropriate message like "Please join the group to comment."
3. Have checkbox allowing admins to decide which one they'd prefer.

The PR is now doing (2), but after discussing with my client, it's too noisy so we need to be able to turn off these messages.  So I'd be fine with (1) or (3).

* [ ] Choose a messaging type (show a message, don't, or make it configurable)

## Issue tracker

["Please log in or sign up to comment" is displayed to authenticated users](https://www.drupal.org/project/social/issues/3280216)

## How to test

1. [ ] Create an open group.
1. [ ] Add group members.
1. [ ] Have group members add some content.
1. [ ] Create a user without placing her in the group.
1. [ ] Have her log in and view the content.
1. [ ] She should not see: "Please log in or sign up to comment".

## Definition of done
### Before merge

### After merge

## Screenshots
*[Required if new feature, and if applicable] If this Pull Request makes visual changes then please include some screenshots that show what has changed here. A before and after screenshot helps the reviewer determine what changes were made.*

## Release notes
*[Required if new feature, and if applicable] A short summary of the changes that were made that can be included in release notes.*

## Change Record
*[Required if applicable] If this Pull Request changes the way that developers should do things or introduces a new API for developers then a change record to document this is needed. Please provide a draft for a change record or a link to an unpublished change record below. Existing change records can be consulted as example. Please provide a draft for a change record or a link to an unpublished change record below. [Existing change records](https://www.drupal.org/list-changes/social) can be consulted as example.*

## Translations
*[Optional]Translatable strings are always extracted from the latest development branch. To ensure translations remain available for platforms running older versions of Open Social the original string should be added to `translations.php` when it's changed or removed.*
